### PR TITLE
Update the software environment and compiler versisons on Summit

### PR DESCRIPTION
**Pull request by minxu74**
After recent Summit update, some old versions of the software environment and compilers have been removed. So this PR is to update them to the new default versions or versions that have been tested on Summit.  Main updates include:

- update PGI to nvhpc/21.3
- update GNU to 9.1.0
- update IBM to 16.1.1-10
- update spectrum-mpi to 10.4.0.3-20210112
- update the '--nrs' parameter for scm. Thanks Azamat.

Some e3sm_integration test suite  results (total(fail)):
- GNU: build: 95(0), run: 95(14)
```
    FAIL ERP_Ld3.ne30pg2_EC30to60E2r2.WCYCL1850.summit_gnu.allactive-pioroot1 RUN time=135
    FAIL ERS.ne11_oQU240.A_WCYCL1850.summit_gnu RUN time=51
    FAIL NCK.ne11_oQU240.A_WCYCL1850.summit_gnu RUN time=51
    FAIL PEM_Ln9.ne30pg2_EC30to60E2r2.WCYCL1850.summit_gnu RUN time=125
    FAIL PET_Ln9_PS.ne30pg2_EC30to60E2r2.WCYCL1850.summit_gnu.allactive-mach-pet RUN time=61
    FAIL SMS.T62_oQU120_ais20.MPAS_LISIO_TEST.summit_gnu RUN time=25
    FAIL SMS_D_Ld1.ne30pg2_EC30to60E2r2.WCYCL1850.summit_gnu.allactive-wcprod RUN time=5883
    FAIL SMS_D_Ld1.ne30pg2_r05_EC30to60E2r2.A_WCYCL1850S_CMIP6.summit_gnu RUN time=6000
    FAIL SMS_D_Ln5.conusx4v1_conusx4v1.FC5AV1C-L.summit_gnu RUN time=2160
    FAIL SMS_D_Ln5.ne4_ne4.FC5AV1C-L.summit_gnu.eam-clubb_sp RUN time=78
    FAIL SMS_Ld2.ne30_oECv3.BGCEXP_CNTL_CNPECACNT_1850.summit_gnu.elm-bgcexp RUN time=126
    FAIL SMS_Ld2.ne30_oECv3.BGCEXP_CNTL_CNPRDCTC_1850.summit_gnu.elm-bgcexp RUN time=126
    FAIL SMS_P12x2.ne4_oQU240.A_WCYCL1850.summit_gnu.allactive-mach_mods RUN time=41
    FAIL SMS_R_Ld5.ne4_ne4.FSCM5A97.summit_gnu.eam-scm RUN time=566
```
- GNUGPU: build: 64(0), run: 64(10)
```
    FAIL NCK.ne11_oQU240.A_WCYCL1850.summit_gnugpu RUN time=55
    FAIL PEM_Ln9.ne30pg2_EC30to60E2r2.WCYCL1850.summit_gnugpu RUN time=99
    FAIL PET_Ln9_PS.ne30pg2_EC30to60E2r2.WCYCL1850.summit_gnugpu.allactive-mach-pet RUN time=559
    FAIL SMS.T62_oQU120_ais20.MPAS_LISIO_TEST.summit_gnugpu RUN time=23
    FAIL SMS_D_Ld1.ne30pg2_EC30to60E2r2.WCYCL1850.summit_gnugpu.allactive-wcprod RUN time=6486
    FAIL SMS_D_Ld1.ne30pg2_r05_EC30to60E2r2.A_WCYCL1850S_CMIP6.summit_gnugpu RUN time=6781
    FAIL SMS_Ld2.ne30_oECv3.BGCEXP_CNTL_CNPECACNT_1850.summit_gnugpu.elm-bgcexp RUN time=100
    FAIL SMS_Ld2.ne30_oECv3.BGCEXP_CNTL_CNPRDCTC_1850.summit_gnugpu.elm-bgcexp RUN time=84
    FAIL SMS_P12x2.ne4_oQU240.A_WCYCL1850.summit_gnugpu.allactive-mach_mods RUN time=40
    FAIL SMS_R_Ld5.ne4_ne4.FSCM5A97.summit_gnugpu.eam-scm RUN time=11
```
- IBM: build: 95(1), run: 94: (3)
```
    FAIL ERS_Ld20.f45_f45.IELMFATES.summit_ibm.elm-fates RUN time=131
    FAIL SMS.ne4_ne4.FC5AV1C-L.summit_ibm.eam-thetanh_ftype2 RUN time=181
    FAIL SMS_R_Ld5.ne4_ne4.FSCM5A97.summit_ibm.eam-scm RUN time=9
```
- IBMGPU: build: 95(1), run: 94: (20)
```
    FAIL ERP_Ld3.ne30pg2_EC30to60E2r2.WCYCL1850.summit_ibmgpu.allactive-pioroot1 RUN time=7199
    FAIL ERP_Ln9.ne4pg2_ne4pg2.F-MMFXX.summit_ibmgpu.eam-mmf_fixed_subcycle RUN time=7205
    FAIL ERS.ne11_oQU240.A_WCYCL1850.summit_ibmgpu RUN time=7198
    FAIL ERS_Ld20.f45_f45.IELMFATES.summit_ibmgpu.elm-fates RUN time=129
    FAIL ERS_Ln9.ne4pg2_ne4pg2.F-MMF1-RCEMIP.summit_ibmgpu RUN time=1864
    FAIL ERS_Ln9.ne4pg2_ne4pg2.F-MMF1.summit_ibmgpu.eam-mmf_crmout RUN time=1860
    FAIL ERS_Ln9.ne4pg2_ne4pg2.F-MMFXX.summit_ibmgpu.eam-mmf_use_VT RUN time=1859
    FAIL ERS_Ln9_P96x1.ne4pg2_ne4pg2.F-MMFOMP.summit_ibmgpu RUN time=1862
    FAIL ERS_Ln9_P96x1.ne4pg2_ne4pg2.F-MMFXX.summit_ibmgpu.eam-mmf_use_ESMT RUN time=1864
    FAIL NCK.ne11_oQU240.A_WCYCL1850.summit_ibmgpu RUN time=7200
    FAIL PEM_Ln9.ne30pg2_EC30to60E2r2.WCYCL1850.summit_ibmgpu RUN time=7198
    FAIL PET_Ln9_PS.ne30pg2_EC30to60E2r2.WCYCL1850.summit_ibmgpu.allactive-mach-pet RUN time=1861
    FAIL SMS.T62_oQU120_ais20.MPAS_LISIO_TEST.summit_ibmgpu RUN time=2705
    FAIL SMS_D_Ld1.T62_oEC60to30v3.DTESTM.summit_ibmgpu RUN time=7202
    FAIL SMS_D_Ld1.ne30pg2_EC30to60E2r2.WCYCL1850.summit_ibmgpu.allactive-wcprod RUN time=7198
    FAIL SMS_D_Ld1.ne30pg2_r05_EC30to60E2r2.A_WCYCL1850S_CMIP6.summit_ibmgpu RUN time=7198
    FAIL SMS_Ld2.ne30_oECv3.BGCEXP_CNTL_CNPECACNT_1850.summit_ibmgpu.elm-bgcexp RUN time=7201
    FAIL SMS_Ld2.ne30_oECv3.BGCEXP_CNTL_CNPRDCTC_1850.summit_ibmgpu.elm-bgcexp RUN time=7200
    FAIL SMS_P12x2.ne4_oQU240.A_WCYCL1850.summit_ibmgpu.allactive-mach_mods RUN time=1860
    FAIL SMS_R_Ld5.ne4_ne4.FSCM5A97.summit_ibmgpu.eam-scm RUN time=12
```

- nvhpc, so many run-time error. see issue #4479

### CODE REVIEW

File: cime_config/machines/config_machines.xml [hunk 1]
```diff
@@ -2850,73 +2849,72 @@
...
+      <init_path lang="perl">/sw/summit/lmod/8.4/init/perl</init_path>
       <cmd_path lang="perl">module</cmd_path>
-      <cmd_path lang="python">/sw/summit/lmod/7.7.10/rhel7.3_gnu4.8.5/lmod/7.7.10/libexec/lmod python</cmd_path>
+      <cmd_path lang="python">/sw/summit/lmod/8.4/libexec/lmod python</cmd_path>
       <cmd_path lang="sh">module</cmd_path>
       <cmd_path lang="csh">module</cmd_path>
       <modules>
         <command name="purge"/>
-        <command name="ls"/>
+        <command name="list"/>
```
Comment (amametjanov) [inline cime_config/machines/config_machines.xml:2862]: There is no use listing modules right after `purge`: might as well either remove `list` here or move it to the end (to show which modules were loaded).

File: cime_config/machines/config_machines.xml [hunk 2]
```diff
@@ -2850,73 +2849,72 @@
...
+        <command name="load">git/2.31.1</command>
         <command name="load">cmake/3.20.2</command>
-        <command name="load">essl/6.1.0-2</command>
+        <command name="load">essl/6.3.0</command>
         <command name="load">netlib-lapack/3.8.0</command>
       </modules>
       <modules compiler="pgi.*">
-        <command name="load">pgi/19.9</command>
-        <command name="load">pgi-cxx14/default</command>
+        <command name="load">nvhpc</command>
```
Comment (amametjanov) [inline cime_config/machines/config_machines.xml:2872]: Can you add `nvhpc` version? Default versions change after PM days and loading specific version might help reproducibility.

File: cime_config/machines/config_machines.xml [hunk 3]
```diff
@@ -2850,73 +2849,72 @@
...
         <command name="load">cuda/10.1.243</command>
       </modules>
       <modules compiler="gnu">
-        <command name="load">gcc/8.1.1</command>
+        <command name="load">gcc/9.1.0</command>
       </modules>
       <modules compiler="gnugpu">
-        <command name="load">gcc/8.1.1</command>
+        <command name="load">gcc/9.1.0</command>
       </modules>
```
Comment (amametjanov) [inline cime_config/machines/config_machines.xml:2891]: Please combine into just single `gnu.*` block.

File: cime_config/machines/config_machines.xml [hunk 4]
```diff
@@ -2850,73 +2849,72 @@
...
+        <command name="load">spectrum-mpi/10.4.0.3-20210112</command>
       </modules>
       <modules compiler="gnu" mpilib="!mpi-serial">
-        <command name="load">spectrum-mpi/10.3.1.2-20200121</command>
+        <command name="load">spectrum-mpi/10.4.0.3-20210112</command>
       </modules>
       <modules compiler="gnugpu" mpilib="!mpi-serial">
-        <command name="load">spectrum-mpi/10.3.1.2-20200121</command>
+        <command name="load">spectrum-mpi/10.4.0.3-20210112</command>
       </modules>
```
Comment (amametjanov) [inline cime_config/machines/config_machines.xml:2907]: Please combine into just single `<modules>` block: same mpi versions for all. Also, i don't think `mpi-serial` is used anywhere anymore.

**Review by amametjanov — COMMENTED** [review]
Adding
```
<MAX_TASKS_PER_NODE compiler="ibmgpu">42</MAX_TASKS_PER_NODE>
<MAX_MPITASKS_PER_NODE compiler="ibmgpu">42</MAX_MPITASKS_PER_NODE>
```
should help with any run-time allocation errors in seaice+coupled OpenMP-offload GPU runs.

**Comment by minxu74** [comment]
Update: 

For gnu and gnugpu, the following two tests failed because the running time exceeded the job wall clock time. Thanks @whannah1  
```
 SMS_D_Ld1.ne30pg2_EC30to60E2r2.WCYCL1850.summit_gnu.allactive-wcprod
 SMS_D_Ld1.ne30pg2_r05_EC30to60E2r2.A_WCYCL1850S_CMIP6.summit_gnu
```
For ibmgpu, the above tests failed with the same reasons. But there was no output in the e3sm.log file except for one error message. It seemed that the program has hung there for 2 hours without doing anything.  @amametjanov can you look at it?

**Review by whannah1 — APPROVED** [review]

**Review by amametjanov — APPROVED** [review]

**Pull request merged**